t80s_reduce/t80s_sort_night/sort_night_files.py

```diff
@@ -0,0 +1,208 @@
+                                    str(frame['FILENAME']) for frame in cursor]
+        return self
+
+    def querySkyFlat(self):
+        self.log.debug('Querying for sky-flat images.')
+        cursor = self.cursor({'IMAGETYP': 'sky-flat'}, [('FILENAME', 1), ])
+        self.log.debug('Found %i sky-flat frames this night.' % cursor.count())
+
+        infos = array([(str(entry['FILENAME']),
+                           str(entry['FILTER'])) for entry in cursor]).T
+
+        ufilters = [str(flt) for flt in unique(infos[1])]
+
+        targets['calibrations']['sky-flat']['filters'] = ufilters
+
+        for filter in ufilters:
+            self.targets['calibrations']['sky-flat'][filter] = { 'night' :
+                                                                night }
+            mask = infos[1] == filter
+            raw = [str(ff) for ff in infos[0][mask]]
+            self.targets['calibrations']['sky-flat'][filter]['raw files'] = raw
+        return self
+
+    def saveFile(self):
+        self.log.info('Saving selection to %s' % self.args.output)
+        with open(self.args.output, 'w') as fp:
+            yaml.dump(self.targets, fp, default_flow_style = False)
+
+    def queryFromLocalImages(self):
+
+        if(self.args.input is None):
+            return self
+
+        files = glob.glob(self.args.input + "/*.fits")
+
+        if(len(files) == 0):
+            self.log.debug("No objects found in %s" % self.args.input)
+            raise NameError("No objects in the Directory %s")
+
+        self.log.debug('Found %i objects,' % len(files))
+        self.log.debug('in the directory %s objects.' % self.args.input)
+        i = 0
+        self.targets = {'night' : self.night,
+            'objects': {}}
+        teste = []
+        raws = {}
+
+        for obj in files:
+            hdulist = fits.open(obj)
+            header = hdulist[0].header
+            raw = obj.split("/")[-1]
+
+            if('OBJECT' in header.keys()):
+                self.log.debug('%s' % header['OBJECT'])
+                if(header['OBJECT'] not in teste):
+                    teste.append(header['OBJECT'])
+                    self.targets['objects'][header['OBJECT']] = {'type': 'UNKNOWN',
+                                                'night': self.night}
+                    self.targets['objects'][header['OBJECT']][
+                        header["FILTER"]] = { 'raw files': [raw]}
+                else:
+                    if(header["FILTER"] in self.targets['objects'][header['OBJECT']].keys()):
+                        self.targets['objects'][header['OBJECT']][
+                        header["FILTER"]]['raw files'].append(raw)
+                    else:
+                        self.targets['objects'][header['OBJECT']][
+                        header["FILTER"]] = { 'raw files': [raw]}
+
+        return self
+
+
+def main(argv):
+    SortNightFiles(argv).queryFromLocalImages().saveFile()
+
+
+if(__name__ == "__main__"):
```

Comment:
This needs to be worked out a little bit, but the overall code is pretty neat. Let's review it latter, once we go through the basic stuff.